api/src/main/java/org/pmiops/workbench/wsm/WsmClient.java

```diff
@@ -0,0 +1,153 @@
+   *
+   * @param sourceWorkspaceUUID UUID of the source workspace
+   * @param targetWorkspace Target workspace object
+   * @return WorkspaceDescription of the cloned workspace
+   */
+  public WorkspaceDescription cloneWorkspaceAsService(
+      String sourceWorkspaceUUID, Workspace targetWorkspace) {
+
+    return wsmRetryHandler.run(
+        context -> {
+          CloneWorkspaceV2Request cloneWorkspaceV2Request =
+              new CloneWorkspaceV2Request()
+                  .userFacingId(targetWorkspace.getNamespace())
+                  .displayName(targetWorkspace.getName())
+                  .additionalProperties(
+                      stringMapToProperties(
+                          Map.of(
+                              "terra-workspace-short-description",
+                              targetWorkspace.getResearchPurpose().getOtherPurposeDetails())))
+                  .description(formatWorkspaceDescription(targetWorkspace))
+                  .cloudResourceGroupId(workbenchConfigProvider.get().vwb.defaultPodId)
+                  .organizationId(workbenchConfigProvider.get().vwb.organizationId)
+                  .jobControl(new JobControl().id(UUID.randomUUID().toString()));
+
+          CloneWorkspaceV2Result cloneWorkspaceV2Result =
+              workspaceServiceApi
+                  .get()
+                  .cloneWorkspaceV2(cloneWorkspaceV2Request, sourceWorkspaceUUID);
+
+          // call the get workspace endpoint to get the full description object
+          return workspaceServiceApi
+              .get()
+              .getWorkspace(
+                  cloneWorkspaceV2Result.getCloneResult().getDestinationWorkspaceId().toString(),
+                  null);
+        });
+  }
+
+  public void shareWorkspaceAsService(String workspaceId, String userEmail, IamRole role) {
+    wsmRetryHandler.run(
+        context -> {
+          GrantRoleRequestBody grantRoleRequestBody =
+              new GrantRoleRequestBody().memberEmail(userEmail);
+          workspaceServiceApi.get().grantRole(grantRoleRequestBody, workspaceId, role);
+          return null;
+        });
+  }
+
+  /**
+   * Waits for the workspace creation to complete.
+   *
+   * @param workspaceId ID of the workspace
+   * @throws InterruptedException if the thread is interrupted
+   * @throws ApiException if there is an API error
+   */
+  public void waitForWorkspaceCreation(String workspaceId)
+      throws InterruptedException, ApiException {
+    int retries = 0;
+    State state = State.CREATING;
+    // Wait for 2 minutes for the workspace creation before giving up
+    do {
+      WorkspaceDescription workspaceDescription =
+          workspaceServiceApi.get().getWorkspace(workspaceId, null);
+      state = workspaceDescription.getOperationState().getState();
+      logger.debug("Workspace state: {}", state);
+      if (state == State.READY) {
+        return;
+      }
+      retries++;
+      Thread.sleep(10_000);
```

Comment:
optional nit - easier to read version of polling: 
https://github.com/all-of-us/workbench/blob/c4f0957478f2dbf3c2ba15a769344a30134a4d8a/api/src/main/java/org/pmiops/workbench/google/CloudBillingClientImpl.java#L32L54